### PR TITLE
[Button] Design guidelines

### PR DESCRIPTION
**Pull request by josepmartins**
## Summary

Add `Button` design guidelines.

## List of notable changes:

- **added** new ComponentLayout to `Button` docs
- **added** design guidelines
- **removed** duplicated documentation

## What should reviewers focus on?

- 👀 Documentation: https://primer-d2bacb09d6-26139705.drafts.github.io/components/Button

## Supporting resources (related issues, external links, etc):

- Epic: https://github.com/github/primer/issues/2231
- Tracking issue: https://github.com/github/primer/issues/1979

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

## Screenshots:

> Please try to provide before and after screenshots or videos

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

 </td>
<td valign="top">

</td>
</tr>
</table>

### CODE REVIEW

File: apps/docs/content/components/Button/index.mdx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+#### Secondary
+
+Secondary buttons are the default button type. Use them for the main actions on a page or form. Secondary buttons are the most common button type and should be used in most cases. They can be paired with a primary button to perform a secondary action, or used on their own.
+
+![An image showing all secondary states.](https://github.com/primer/brand/assets/912236/1acc8007-13a2-4083-b0f2-c1b4d5cf7b82)
+
+#### Subtle
+
+Subtle buttons have a transparent background with translucent hover and active states, making them useful for compound components like [Button Group](/components/ButtonGroup) when paired with secondary or primary variants.
```
Comment (danielguillan) [inline apps/docs/content/components/Button/index.mdx:72]: Should we also note that the subtle variant is tertiary in terms of hierarchy?

File: apps/docs/content/components/Button/index.mdx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+![An image showing all subtle states.](https://github.com/primer/brand/assets/912236/7420969a-e561-4fd6-9bef-7340e2b0a7e8)
+
+### Trailing and leading icons
+
+You can place an icon inside the Button in either the leading or the trailing position to enhance the visual context. It is recommended to use an use an [Octicon](https://primer.style/octicons) here.
```
Comment (danielguillan) [inline apps/docs/content/components/Button/index.mdx:78]:
```suggestion
You can place an icon inside the Button in either the leading or the trailing position to enhance the visual context. It is recommended to use an [Octicon](https://primer.style/octicons) here.
```

File: apps/docs/content/components/Button/index.mdx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+![An image showing all secondary states.](https://github.com/primer/brand/assets/912236/1acc8007-13a2-4083-b0f2-c1b4d5cf7b82)
+
+#### Subtle
+
+Subtle buttons have a transparent background with translucent hover and active states, making them useful for compound components like [Button Group](/components/ButtonGroup) when paired with secondary or primary variants.
+
+![An image showing all subtle states.](https://github.com/primer/brand/assets/912236/7420969a-e561-4fd6-9bef-7340e2b0a7e8)
+
+### Trailing and leading icons
```
Comment (danielguillan) [inline apps/docs/content/components/Button/index.mdx:76]:
```suggestion
### Leading and trailing icons
```

File: apps/docs/content/components/Button/index.mdx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  <Do>
+    <img src="https://github.com/primer/brand/assets/912236/2de44a58-fbf6-46bf-a645-5efd88da70dd" />
+    <Caption>
+      Use a leading icon that provides context and reinforces the label.
+    </Caption>
+  </Do>
+  <Dont>
+    <img src="https://github.com/primer/brand/assets/912236/9dba03c8-5a70-464c-9100-33a2205e4c26" />
+    <Caption>
+      Don't use a trailing icons to add a visual cue to the label. Use the
```
Comment (danielguillan) [inline apps/docs/content/components/Button/index.mdx:92]:
```suggestion
      Don't use a trailing icon to add a visual cue to the label. Use the
```

File: apps/docs/content/components/Button/index.mdx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+Button default to a medium size. Medium is suitable for most interfaces and is considered the preferred size.
+
+Large buttons increase the significance of an action and should be used sparingly. Use the large option when you need to emphasize the button on a big sized heading, for example in a [Hero](/components/Hero) component.
+
+### Arrow
+
+The arrow is shown by default in all buttons. Hide the arrow only to save horizontal space or to diminish the importance of the button.
+
+When used in a group, the arrow should be used or hide consistently across all buttons.
```
Comment (danielguillan) [inline apps/docs/content/components/Button/index.mdx:125]:
```suggestion
When used in a group, the arrow should be used or hidden consistently across all buttons.
```

File: apps/docs/content/components/Button/index.mdx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+### Size
+
+Button default to a medium size. Medium is suitable for most interfaces and is considered the preferred size.
+
+Large buttons increase the significance of an action and should be used sparingly. Use the large option when you need to emphasize the button on a big sized heading, for example in a [Hero](/components/Hero) component.
+
+### Arrow
+
+The arrow is shown by default in all buttons. Hide the arrow only to save horizontal space or to diminish the importance of the button.
```
[the remark this answers is not in the document]
Comment (josepmartins) [reply apps/docs/content/components/Button/index.mdx:123]: Good point! I'll add that 👍

**Review by danielguillan — APPROVED** [review]

File: apps/docs/content/components/Button/index.mdx [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+---
+
+import ComponentLayout from '../../../src/layouts/component-layout'
+export default ComponentLayout
+
+![An image showing different Button types.](https://github.com/primer/brand/assets/912236/435f104b-47a5-49d8-9568-6825f2b3789c)
+
+## Usage
+
+Buttons allow users to trigger an action or navigate to a new page when clicked. The button's label leads with an actionable verb, clearly indicating its purpose. Use the "Default" button variant primarily, and only reach for the other variants for hierarchical or functional purposes.
```
Comment (mperrotti) [inline apps/docs/content/components/Button/index.mdx:13]: We should mention that the button should be rendered as `<a>` not `<button>` it it's used to navigate to a new page.

**Review by mperrotti — APPROVED** [review]

File: apps/docs/content/components/Button/index.mdx (hunk 7 above)
Comment (josepmartins) [reply apps/docs/content/components/Button/index.mdx:13]: Good point! I'll add that note

**Pull request merged**